src/librustc_interface/passes.rs

```diff
@@ -597,21 +597,9 @@ fn write_out_deps(
         // deleted. For more info see #28735
         for path in files {
             //FIXME: async!
-            let mut reader = io::BufReader::new(fs::File::open(&path)?);
-            let mut hasher = rustc_data_structures::fx::FxHasher::default();
-            let mut buffer = [0; 1024];
-            loop {
-                let count = reader.read(&mut buffer)?;
-                if count == 0 {
-                    break;
-                }
-                hasher.write(&buffer[..count]);
-            }
-            let hash = hasher.finish();
-            let size = std::fs::metadata(&path)?.len();
-
+            let (size, hash) = hash_src(&path).unwrap_or((0, 0));
             writeln!(file, "{}:", path)?;
-            writeln!(file, "# size:{} hash:{}", size, hash)?;
+            writeln!(file, "# size:{} hash:{}", size as u32, hash)?;
```

Comment:
size is already an u32

Reply:
reader.read returns usize - I was just saving the lower 32bits. Let me change it here to write the actual size out - I think that's less confusing. If cargo wants to do optimisations it can do that independently.